include/deal.II/base/tensor.h

```diff
@@ -205,6 +205,16 @@ class Tensor<0,dim,Number>
   template <typename OtherNumber>
   Tensor<0,dim,Number> &operator = (const Tensor<0,dim,OtherNumber> &rhs);
 
+  /**
+   * Assignment from tensors with same underlying scalar type.
+   */
+  Tensor<0,dim,Number> &operator = (const Tensor<0,dim,Number> &rhs);
+
+  /**
+   * Remove non-const default assignment operator to avoid ambiguity.
+   */
+  Tensor<0,dim,Number> &operator = (Tensor<0,dim,Number> &) = delete;
```

Comment:
Hm, can you explain why you need this?

Reply:
Because the default operator doesn't work with `const` parameter and the corresponding error is triggered until the user operator is defined, I guess the compiler might provide some kinds of non-const assignment operator. To avoid any uncontrollable behavior, I marked this function as `deleted`.